pydicom/filewriter.py

```diff
@@ -145,13 +146,15 @@ def correct_ambiguous_vr(ds, is_little_endian):
         The corrected dataset
     """
     # Iterate through the elements
-    for elem in ds:
-        # Iterate the correction through any sequences
-        if elem.VR == 'SQ':
-            for item in elem:
-                item = correct_ambiguous_vr(item, is_little_endian)
-        elif 'or' in elem.VR:
-            elem = correct_ambiguous_vr_element(elem, ds, is_little_endian)
+    if isinstance(ds, Dataset):
```

Comment:
Do you need the type check? I don't think there'd be ambiguous elements in FileDataset or DicomDir

Reply:
You are right - the ambiguous handling may be problematic. 
My reasoning was that the conversion is only needed if going from implicit to explicit VR, and in this case I convert the data element anyway from raw. What I did not do in this case was the conversion for ambiguous elements, and also I did not update the dataset with the converted elements, which means it has to be converted again if needed.

I hadn't thought this through as @darcymason was to change this anyway, but as he isn't going to do it probably, I have to check again. Currently I come to the same conclusion - I need the dataset in the `write_data_element`. I will check if I can find a better solution... otherwise this would be an API change.